Sources/StreamChat/Controllers/MessageController/MessageController.swift

```diff
@@ -722,19 +722,77 @@ public class ChatMessageController: DataController, DelegateCallable, DataStoreP
     }
 
     /// Marks the thread read if this message is the root of a thread.
-    internal func markThreadRead(completion: ((Error?) -> Void)? = nil) {
+    public func markThreadRead(completion: ((Error?) -> Void)? = nil) {
         messageUpdater.markThreadRead(cid: cid, threadId: messageId) { error in
-            completion?(error)
+            self.callback {
+                completion?(error)
+            }
         }
     }
 
     /// Marks the thread unread if this message is the root of a thread.
-    internal func markThreadUnread(completion: ((Error?) -> Void)? = nil) {
+    public func markThreadUnread(completion: ((Error?) -> Void)? = nil) {
         messageUpdater.markThreadUnread(
             cid: cid,
             threadId: messageId
         ) { error in
-            completion?(error)
+            self.callback {
+                completion?(error)
+            }
+        }
+    }
+
+    /// Fetches the thread information of the message this controller manages.
+    /// Returns an error in case the message is not the root of a thread.
+    /// - Parameters:
+    ///   - replyLimit: The number of replies fetched.
+    ///   - participantLimit: The number of participants fetches.
```

Comment:
completion missing here as well

Reply:
Same as above, I think that completions are kinda repetitive documentation, it is pretty clear what they do 🤔